vllm/v1/core/single_type_kv_cache_manager.py

```diff
@@ -292,15 +280,37 @@ def remove_skipped_blocks(self, request_id: str,
         # No need to remove blocks for full attention.
         pass
 
-    def get_num_common_prefix_blocks(self, request_id: str,
-                                     num_running_requests: int) -> int:
-        blocks = self.req_to_blocks[request_id]
+    def get_num_common_prefix_blocks(
+            self, running_request_id: str, running_request_ids: list[str],
+            transfering_request_ids: list[str]) -> int:
+        """Get common prefix blocks shared by all running and transferring 
+        requests."""
+        if running_request_id not in self.req_to_blocks:
+            return 0
+
+        reference_blocks = self.req_to_blocks[running_request_id]
+        total_requests = len(running_request_ids) + len(
+            transfering_request_ids)
+
+        transferring_blocks = [
+            self.req_to_blocks[req_id] for req_id in transfering_request_ids
+            if req_id in self.req_to_blocks
+        ]
+
         num_common_blocks = 0
-        for block in blocks:
-            if block.ref_cnt == num_running_requests:
-                num_common_blocks += 1
-            else:
+        for i, ref_block in enumerate(reference_blocks):
+
+            if ref_block.ref_cnt < total_requests:
+                break
+
+            transferring_has_block = sum(
+                1 for blocks in transferring_blocks if i < len(blocks)
+                and blocks[i].block_id == ref_block.block_id)
+
+            if transferring_has_block != len(transfering_request_ids):
```

Comment:
I'm thinking of:
```
if ref_block.ref_cnt - transferring_has_block == len(running_request_id): 
    # it is a common block
else:
    # it is not a common block
```
is it correct?

Reply:
Hi @heheda12345 I think this makes sense. It is also simpler and there is no significant performance overhead from this. I have implemented this in the new commit I pushed